### PR TITLE
[main] Use openshift-knative/hack/cmd/generate for generating Dockerfiles

### PR DESCRIPTION
**Pull request by mgencur**
**What this PR does / why we need it**:

This is a variant of https://github.com/openshift-knative/serving/pull/832 for "main" branch.
Should be merged only after pull/832 is approved.

**Which issue(s) this PR fixes**:

JIRA: https://issues.redhat.com/browse/SRVKS-1265

**Does this PR needs for other branches**:

**Does this PR (patch) needs to update/drop in the future?**:

JIRA:

### CODE REVIEW

**Comment by mgencur** [comment]
/hold

For https://github.com/openshift-knative/serving/pull/832 to be approved.

**Comment by skonto** [comment]
@mgencur what is the diff compared to https://github.com/openshift-knative/serving/pull/832 should I test against release-next?

**Comment by skonto** [comment]
I am going to merge test PR passes.

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 CGO_ENABLED=0
 GOOS=linux
-CORE_IMAGES=./cmd/activator ./cmd/autoscaler ./cmd/autoscaler-hpa ./cmd/controller ./cmd/queue ./cmd/webhook ./vendor/knative.dev/pkg/apiextensions/storageversion/cmd/migrate
+CORE_IMAGES=./cmd/activator ./cmd/autoscaler ./cmd/autoscaler-hpa ./cmd/controller ./cmd/queue ./cmd/webhook ./cmd/default-domain ./pkg/cleanup/cmd/cleanup ./vendor/knative.dev/pkg/apiextensions/storageversion/cmd/migrate
```
Comment (skonto) [inline Makefile:5]: We might want to skip cleanup or remove the logic we have in S-O for resource cleanup cc @dsimansk, not going to block this PR.
Also I think we have the same in 1.15.

**Comment by skonto** [comment]
Test PR passes I will merge.

**Pull request merged**